.github/workflows/callable_swap_tests.yml

```diff
@@ -0,0 +1,114 @@
+name: Callable Swap Functional tests
+
+on:
+  workflow_call:
+    inputs:
+      app_build_artifact:
+        required: true
+        type: string
+      exchange_build_artifact:
+        required: false
+        default: ''
+        description: 'If not provided, the workflow will build the Exchange app.'
+        type: string
+      ethereum_build_artifact:
+        required: false
+        default: ''
+        description: 'If not provided, the workflow will build the Ethereum app.'
+        type: string
+      regenerate_snapshots:
+        required: false
+        default: false
+        type: boolean
+
+env:
+  SWAP_TEST_DIR: "tests/swap"
+
+jobs:
+  build_exchange:
+    name: Build Exchange app using the reusable workflow
+    if: ${{ inputs.exchange_build_artifact == '' }}
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@y333/manage_several_pytest_directories
+    with:
+      app_repository: 'LedgerHQ/app-exchange'
+      app_branch_name: 'develop'
+      upload_app_binaries_artifact: "app_exchange_binaries"
+      use_case: 'dbg_full_replay'
```

Comment:
Can we use `use_test_keys` ? I had issues hidden by the usage of DEBUG once and since I never put it in CI